python/Ganga/GPIDev/Lib/Registry/RegistrySlice.py

```diff
@@ -304,7 +301,8 @@ def copy(self, keep_going):
         return this_slice
 
     def __contains__(self, j):
-        return j.id in self.objects.keys()
+        allObjK = self.objects.keys()
+        return j.id in allObjK
```

Comment:
This was just a comment as to why we can't drop the use of some ```.keys``` calls, but I hope this is clear in the revision after this which touches less code.